### ISSUE TITLE
It would be nice to have the JSON logging style optional

### ISSUE THREAD

**Issue by aeimer** [bitnami/bitnami-docker-rabbitmq#177]
**Description**

With rabbitmq `3.9` there is an json logger option:
https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.9.0

I tried different things and were not able to make this work.

- https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/json-log
- https://github.com/rabbitmq/rabbitmq-server/pull/2861
- https://github.com/rabbitmq/rabbitmq-server/pull/2927

My example configs with the bitnami/rabbitmq helmchart v 8.22.4 looks like

```yaml
# We want to use JSON logging; see config
logs: ""

extraConfiguration: |-
  # FROM https://github.com/rabbitmq/cluster-operator/tree/fb8126f7a9ad96962a2318edb3aa5a9301c272ab/docs/examples/json-log
  # FIXME: This should work, but it does not...
  log.console = true
  log.console.level = info
  log.console.formatter = json
  log.console.formatter.json.field_map = verbosity:v time msg domain file line pid level:-
  log.console.formatter.json.verbosity_map = debug:7 info:6 notice:5 warning:4 error:3 critical:2 alert:1 emergency:0
  log.console.formatter.single_line = on
  log.console.formatter.time_format = rfc3339_T
```

I would love to have a flag to enable json via helm or some help in the values.yaml on how to setup and configure this.

**Steps to reproduce the issue:**

1. [First Step]
2. [Second Step]
3. [and so on...]

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
(paste your output here)
```

- Output of `docker info`:

```
(paste your output here)
```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by alvneiayu** [comment]
hi @aeimer 

Sound really great. If you want to send us a contribution including this, we are open to receive PRs. I will be really happy to review it. Please let me know.

Thanks a lot

Álvaro

**Comment by alvneiayu** [comment]
hi @aeimer 

First of all, sorry for the late reply. Mmmmm it is so strange, I can not reproduce your problem. I can pull the image and it is running setting up the same image that you are having problems (image: docker.io/bitnami/rabbitmq:3.9.7-debian-10-r0).

What is the problem that you are reproducing? Pulling problems?

Thanks a lot and sorry again for the late reply.

Álvaro

**Comment by aeimer** [comment]
Hi @alvneiayu,

I am able to download and run both images.
But when I pass the config file the logs are not output in json (like its configured). On the stock rabbitmq the config works neat.

So my question is, why does the rabbitmq not load that config file. I explcitly looked into the helm chart and the documentation of that image, I was not able to find a proper answer to this question. I don't know why that conf file has no affect in the image.
Is there any entrypoint.sh magic or sth? The helm chart just overwrites the `/bitnami/rabbitmq/conf/rabbitmq.conf`, so did I to test the changes:
![image](https://user-images.githubusercontent.com/25035182/137089591-31712e60-203b-4d42-9c5d-7744d9b07afd.png)
![image](https://user-images.githubusercontent.com/25035182/137089668-108d60f2-0f6f-43fd-b787-e5c2d9e2cbe0.png)

**Comment by anthonyhaussman** [comment]
I think to have found the problem to setup JSON logs.
Configuring `extraConfiguration` to add log config is the right way to go by checking the RabbitMQ documentation:
```
extraConfiguration: |-
  ...
  log.console.level = info
  log.console.formatter = json
  log.file = false
  ...
```

But the problem comes from the environment variable `RABBITMQ_LOGS`.
To use `log` configuration, we need to unset `RABBITMQ_LOGS` but the `rabbitmq-env.sh` script file launches at the startup set it to `-` if empty:

https://github.com/bitnami/bitnami-docker-rabbitmq/blob/ba8fafaa3470c86f2068f3167da8b17d3d3573bc/3.8/debian-10/rootfs/opt/bitnami/scripts/rabbitmq-env.sh#L154-L155

If `RABBITMQ_LOGS` environment is set, it prevails on the `log` configuration.
In conclusion, the chart/code needs to be updated in adequation to use JSON logs from rabbitmq-bitnami.

**Comment by carrodher** [comment]
Thanks for reporting this issue. Would you like to contribute by creating a PR to solve the issue? The Bitnami team will be happy to review it and provide feedback. [Here](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md) you can find the contributing guidelines.

**Comment by aeimer** [comment]
@anthonyhaussman do we actually need to `unset` the variable or is it enough to set an empty string `RABBITMQ_LOGS=""`?

**Comment by anthonyhaussman** [comment]
> ```yaml
>   log.console.level = info
>   log.console.formatter = json
>   log.file = false
> ```

@aeimer Set an empty string `RABBITMQ_LOGS=""` works :ok_hand:

**Comment by anthonyhaussman** [comment]
I can confirm with the image `3.9.16-debian-10-r0`, JSON logging via the Bitnami RabbitMQ Helm chart is working. :ok_hand: 

Test settings:
```
...
  logs: ""

  extraConfiguration: |-
    log.console.level = info
    log.console.formatter = json
    log.file = false
...
```

Console output:
```
> kubectl -n rabbitmq logs rabbitmq-ha-0

...
2022-04-28 07:47:04.894236+00:00 [noti] <0.229.0> Logging: switching to configured handler(s); following messages may not be visible in this log output
{"time":"2022-04-28 07:47:04.913915+00:00","level":"notice","msg":"Logging: configured log handlers are now ACTIVE","domain":"rabbitmq.prelaunch","file":"rabbit_prelaunch_logging.erl","line":1546,"mfa":["rabbit_prelaunch_logging","install_handlers",0],"pid":"<0.229.0>"}
{"time":"2022-04-28 07:47:08.808831+00:00","level":"notice","msg":"Application mnesia exited with reason: stopped","domain":"otp","file":"application_controller.erl","line":2119,"mfa":["application_controller","info_exited",3],"pid":"<0.44.0>"}
...
```

**Comment by carrodher** [comment]
Thanks for the contribution and for confirming it worked. I'm closing this issue, feel free to reopen it or create a new one if something doesn't work as expected.

**Comment by aeimer** [comment]
I can also confirm this works! Great job guys, thank you :)
